tokio/src/runtime/metrics/metrics.rs

```diff
@@ -0,0 +1,68 @@
+//! This file contains the types necessary to collect various types of metrics.
+use crate::loom::sync::atomic::{AtomicU64, Ordering};
+
+pub(crate) struct RuntimeMetrics {
+    workers: Box<[WorkerMetrics]>,
+}
+
+pub(crate) struct WorkerMetrics {
+    park_count: AtomicU64,
+    steal_count: AtomicU64,
+    poll_count: AtomicU64,
+}
+
+
+impl RuntimeMetrics {
+    pub(crate) fn new(worker_threads: usize) -> Self {
+        let mut workers = Vec::with_capacity(worker_threads);
+        for _ in 0..worker_threads {
+            workers.push(WorkerMetrics {
+                park_count: AtomicU64::new(0),
+                steal_count: AtomicU64::new(0),
+                poll_count: AtomicU64::new(0),
+            });
+        }
+
+        Self {
+            workers: workers.into_boxed_slice(),
```

Comment:
What do you gain frrom this compared to keeping it a `Vec`?

Reply:
A boxed slice is simpler than a vector, e.g. it doesn't have both a length and capacity.